src/meta-data.ts

```diff
@@ -0,0 +1,28 @@
+import { SpriteProperties, BlockProperties } from './abstracts';
+import { ScratchProject } from './sb-util';
+
+export interface SpriteMeta {
+    project: ScratchProject;
+}
+const spriteMeta = new WeakMap<SpriteProperties, SpriteMeta>();
```

Comment:
All these WeakMaps seem worrying to me. Will this an effect on memory use? Keeping the values up to date may be a lot of work.

Reply:
Maybe it's that this removes an opportunity for an extensible interface for users to produce or help manage their system specific SpriteMeta. Seeing all these singleton instances, I hope we don't need to make them non-singletons later.

Reply:
yeah - i mean - long term i was hoping that the "project" would be the main meta-data-owner - I just am plugging some holes with temporary glue for the time being - and I have the same twinge riding me, the only reason i made separate ones is that I wanted the specific type safety on it, but then i put it in it's own file, and now it probably could be the same map with generic accesors